### ISSUE TITLE
[Bug Report] Bad REINFORCE implementation for Mujoco Tutorial

### ISSUE THREAD

**Issue by JacopoBandoni** [Farama-Foundation/Gymnasium#1336]
### Describe the bug

The tutorial "Training using REINFORCE for Mujoco" does not properly implement the update function as defined in the REINFORCE algorithm.

```python

    def update(self):
        """Updates the policy network's weights."""
        running_g = 0
        gs = []

        # Discounted return (backwards) - [::-1] will return an array in reverse
        for R in self.rewards[::-1]:
            running_g = R + self.gamma * running_g
            gs.insert(0, running_g)

        deltas = torch.tensor(gs)

        log_probs = torch.stack(self.probs)

        # ERROR HERE, why calculate the mean?
        # Calculate the mean of log probabilities for all actions in the episode
        log_prob_mean = log_probs.mean()

        # Update the loss with the mean log probability and deltas
        # Now, we compute the correct total loss by taking the sum of the element-wise products.
        loss = -torch.sum(log_prob_mean * deltas)

        # Update the policy network
        self.optimizer.zero_grad()
        loss.backward()
        self.optimizer.step()

        # Empty / zero out all episode-centric/related variables
        self.probs = []
        self.rewards = []

```

The error in the code occurs because it first computes the **mean** of the log-probabilities and then multiplies it by the vector of discounted returns:

$$
\mathcal{L} = - \sum_{t=0}^{T} G_t \cdot \left( \frac{1}{T} \sum_{t=0}^{T} \log \pi_\theta (a_t | s_t) \right)
$$

Here, the term:

$$
\frac{1}{T} \sum_{t=0}^{T} \log \pi_\theta (a_t | s_t)
$$

represents the **mean** of the log-probabilities over all actions in the episode. This is incorrect because it **loses the correspondence** between each action and its respective discounted return.

The loss should be computed **without taking the mean** of the log-probabilities. Instead, each log-probability should be **directly weighted** by its corresponding discounted return:

$$
\mathcal{L} = - \sum_{t=0}^{T} G_t \log \pi_\theta (a_t | s_t)
$$

Here, each **log-probability** is **individually weighted** by its respective discounted return \( G_t \), ensuring that the correct REINFORCE formulation is followed.

In fact by running the code the algorithm does not learn as shown in the tutorial, I link an image to show the result:

![Image](https://github.com/user-attachments/assets/ba07825a-669c-492f-873f-d6489832094d)

### Code example

```shell

```

### System info

- Installed from source 
- MacOS Sequoia 15.3.1
- Python 3.12.0

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/Gymnasium/issues) in the repo

**Comment by pseudo-rnd-thoughts** [comment]
Thanks for the issue and looking at the implementation so thoughtfully. I'll have a look at your PR
